docs/migration.md

```diff
@@ -0,0 +1,69 @@
+# Migrating ClusterInstallation to Mattermost
+
+As of the new release, the Custom Resource managed by the Mattermost Operator changes from `ClusterInstallation` to `Mattermost`.
+Besides the name change, some new functionality is introduced while other is removed.
+
+`BlueGreen` and `Canary` deployments were not widely used and introduced a lot of complexity, so those features were dropped. In most cases, the multi-replica Mattermost cluster proved to be enough for save updates between versions.
+Behavior similar to `BlueGreen` could be mimicked by using multiple Mattermost resources and with some extra ingresses and services.
+
+## Automatic migration
+It is possible for the Operator to migrate `ClusterInstallation` to `Mattermost` as long as none of the unsupported features are enabled (like `BlueGreen` or `Canary`). 
+During the migration, old Pods are deleted only after the new `Mattermost` resource reaches the `stable` state, 
+therefore **the Mattermost instance should not experience any downtime.**
+
+For migration to be possible, the Mattermost Operator needs to be version `v1.12.x`.
+
+> **NOTE:** Make sure that Mattermost Operator is in version `v1.12.x` before starting the migration.
+> To do that, inspect the Operator image tag. To get full image name and tag, run:
+> ```
+> kubectl -n [OPERATOR_NAMESPACE] get deployment mattermost-operator -o jsonpath='{.spec.template.spec.containers[0].image}'
+> ```
+> If the Operator's minor version is earlier than `12`, update it to `v1.12.0` by updating manifests [to the released version](https://raw.githubusercontent.com/mattermost/mattermost-operator/v1.12.0/docs/mattermost-operator/mattermost-operator.yaml). 
+
+To run the migration, follow the steps:
+
+1. Prepare the namespace and name of the `ClusterInstallation` and export those as environment variables:
```

Comment:
```suggestion
1. Record the namespace and name of the `ClusterInstallation` and export those as environment variables on the device that is performing the migration:
```

Reply:
Changed it slightly, as I feel like it could be a little confusing as the actual migration is performed by the Operator, user only triggers it.

Reply:
Oops, yeah that's what I meant to write. Good catch.